shbin.py

```diff
@@ -60,6 +61,21 @@ def get_repo_and_user():
     return gh.get_repo(os.environ["SHBIN_REPO"]), gh.get_user().login
 
 
+def get_repo(full_name_repo):
+    gh = Github(os.environ["SHBIN_GITHUB_TOKEN"])
+    return gh.get_repo(full_name_repo)
+
+
+def get_path(url):
+    pattern = r"https://github.com/.*?/(?:tree|blob)/(.*)"
+    result = re.findall(pattern, url)
+    if result:
+        extracted_string = result[0]
+        return extracted_string
+    else:
+        raise DocoptExit(f"Ensure your path is from github repository. (error {e})")
```

Comment:
Here `e` in the message is not defined. 

Also, it's considered more pythonic the [EAFP](https://docs.python.org/3/glossary.html#term-EAFP)  approach

So, it should something like 

```suggestion
    try:
        return re.findall(pattern, url)[0]
    except TypeError:
        raise DocoptExit(f"Ensure the URL is from github repository")
```